include/graphene-box2d.h

```diff
@@ -0,0 +1,122 @@
+/* graphene-box2d.h: An 2D, axis aligned bounding box
+ *
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2023  Emmanuele Bassi
+ */
+
+#pragma once
+
+#if !defined(GRAPHENE_H_INSIDE) && !defined(GRAPHENE_COMPILATION)
+#error "Only graphene.h can be included directly."
+#endif
+
+#include "graphene-types.h"
+#include "graphene-vec2.h"
+
+GRAPHENE_BEGIN_DECLS
+
+/**
+ * graphene_box2d_t:
+ *
+ * A 2D box, described as the axis-aligned area between a minimum and
+ * a maximum vertices lying on the same plane.
+ *
+ * Since: 1.12
+ */
+struct _graphene_box2d_t
+{
+  /*< private >*/
+  GRAPHENE_PRIVATE_FIELD (graphene_vec2_t, min);
+  GRAPHENE_PRIVATE_FIELD (graphene_vec2_t, max);
```

Comment:
Should this be a single `graphene_vec4_t` because the vec2's are just vec4's in disguise afaik?

[The Rect type in GSK's Vulkan shader](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/gsk/vulkan/resources/rect.glsl) uses a vec4 itself, and that worked really well.

Reply:
Partial lane reads are always slower than full lane ones; if we store the min/max in the same SIMD type like this:

```
| min_x | min_y | max_x | max_y |
```

then it should be possible to optimise the access using:

```c
graphene_simd4f_t min = graphene_simd4_merge_low (box->value, graphene_sim4f_zero ());
graphene_simd4f_t max = graphene_simd4f_merge_high (box->value, graphene_simd4f_zero ());
```